docs/src/components/BuiltinLegacyRedirect/index.js

```diff
@@ -0,0 +1,13 @@
+import BrowserOnly from "@docusaurus/BrowserOnly";
+import { Redirect, useLocation } from "@docusaurus/router";
+import React from "react";
+
+export default function BuiltinLegacyRedirect({}) {
```

Comment:
I'd be nice to add a short comment on what kinds of links we're handling here. I think we can also put a date like 6 months from now when this component should be removed.